flattentool/sort_iati.py

```diff
@@ -0,0 +1,128 @@
+"""
+Sorts an IATI XML file according to the schema. This is most useful for 2.0x
+where that ordering is required.
+
+Based on https://github.com/OpenDataServices/iati-utils/blob/master/sort_iati.py
+
+Copyright (c) 2013-2014 Ben Webb
+Copyright (c) 2016 Open Data Services Co-operative Limited
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+from collections import OrderedDict
+from warnings import warn
+try:
+    import lxml.etree as ET
+    # Note that lxml is now "required" - it's listed as a requirement in
+    # setup.py and is needed for the tests to pass.
+    # However, stdlib etree still exists as an unsupported feature.
+except ImportError:
+    import xml.etree.ElementTree as ET
+    warn('Using stdlib etree may work, but is not supported. Please install lxml.')
+import sys
+
+# Namespaces necessary for opening schema files
+namespaces = {
+    'xsd': 'http://www.w3.org/2001/XMLSchema'
+}
+
+
+class IATISchemaWalker(object):
+    """
+    Class for converting an IATI XML schema to documentation in the
+    reStructuredText format.
```

Comment:
Would be good to update the doc-string to document what it does do now.

Reply:
Good point! But it’s a slippery slope…!

Reply:
Done in 7322da3… but that led me to fixing a load of other stuff.